project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -1,151 +1,337 @@
-import com.google.appengine.api.datastore.Query.CompositeFilter;
-import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
-import com.google.appengine.api.datastore.Query.Filter;
 import com.google.appengine.api.datastore.Query.FilterOperator;
 import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
 
 
+import java.util.Iterator;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+// import java.net.URL;
+// import java.io.IOException;
+// import java.io.BufferedReader;
+// import java.io.InputStreamReader;
+// import java.net.HttpURLConnection;
+// import javax.servlet.http.HttpServlet;
+// import java.nio.charset.StandardCharsets;
+// import javax.servlet.annotation.WebServlet;
+// import java.io.UnsupportedEncodingException;
+// import org.json.simple.parser.ParseException;
+// import javax.servlet.http.HttpServletRequest;
+// import javax.servlet.http.HttpServletResponse;
+// import com.google.appengine.api.datastore.Query;
+// import com.google.appengine.api.datastore.Entity;
+// import com.google.autograder.servlets.helpers.API;
+// import com.google.autograder.servlets.helpers.Services;
+// import com.google.appengine.api.datastore.Query.Filter;
+// import com.google.appengine.api.datastore.PreparedQuery;
+// import com.google.appengine.api.datastore.Query.SortDirection;
+// import com.google.appengine.api.datastore.Query.FilterOperator;
+// import com.google.autograder.servlets.auth.AccessTokenResponse;
+// import com.google.appengine.api.datastore.Query.FilterPredicate;
+
 /** Class containing basic database functionalities. */
-public class Database {
+public final class Database {
+
+    private static final UserService USER_SERVICE = UserServiceFactory.getUserService();
+    private static final DatastoreService DATA_STORE = DatastoreServiceFactory.getDatastoreService();
+    private static final Gson GSON = new Gson();
+
+    public static final List<String> ANSWER_GRADING_STATUSES = Arrays.asList("NOT_GRADED", "GRADED");
+
+    public static UserService getUserService() {
+        return USER_SERVICE;
+    }
+
+    public static DatastoreService getDataStore() {
+        return DATA_STORE;
+    }
+
+    public static String getCurrentUserEmail() {
+        return USER_SERVICE.getCurrentUser().getEmail();
+    }
+
+    public static String getCurrentUserID() {
+        return USER_SERVICE.getCurrentUser().getUserId();
+    }
+
+    public static PreparedQuery query(Query query) {
+        return DATA_STORE.prepare(query);
+    }
+
+    // Takes in a courses array in JSON format and then stores that courses data for the current user.
+
+    public static void storeCoursesData(String coursesJSON) {
+        try {
```

Comment:
Similar to the other PR, debugging constructs should not be checked in

Reply:
Quick question, what do you mean by debugging constructs here? Do you mean the try-catch?

Reply:
Yes, typically try-catch blocks should be very specific, covering only the lines that you expect might throw an exception. The catch block should also do something with the exception other than logging it, which would happen anyway if you let it propagate to the servlet. Having wide try-catch blocks only impair readability here, as readers are left wondering what lines of code can throw an exception.